.github/workflows/build.yml

```diff
@@ -0,0 +1,30 @@
+name: build
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+permissions:
+  actions: read
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v3
+        with:
+          version: 8
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: pnpm
+      - name: Install dependencies
+        run: pnpm i
+      - name: Build all projects
+        run: pnpm build
```

Comment:
FYI you could simply add this step to your existing `test.yml`. Seems appropriate that part of your testing is if it can build for production.

Reply:
Good call--next pr.